src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenSkillHandler.java

```diff
@@ -0,0 +1,252 @@
+package com.minecolonies.coremod.entity.citizen.citizenhandlers;
+
+import com.google.common.collect.ImmutableMap;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.entity.citizen.AbstractEntityCitizen;
+import com.minecolonies.api.entity.citizen.Skill;
+import com.minecolonies.api.entity.citizen.citizenhandlers.ICitizenSkillHandler;
+import com.minecolonies.coremod.Network;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import com.minecolonies.coremod.network.messages.VanillaParticleMessage;
+import com.minecolonies.coremod.util.ExperienceUtils;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import static com.minecolonies.api.util.constant.CitizenConstants.MAX_CITIZEN_LEVEL;
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_LEVEL_MAP;
+
+/**
+ * The citizen skill handler of the citizen.
+ */
+public class CitizenSkillHandler implements ICitizenSkillHandler
+{
+    /**
+     * Chance to level up intelligence.
+     */
+    private static final int CHANCE_TO_LEVEL = 50;
+
+    /**
+     * Defines how much child stats very from parents average(+ or -).
+     */
+    private static final int CHILD_STATS_VARIANCE = 3;
+
+    /**
+     * Skill map.
+     */
+    public Map<Skill, Tuple<Integer, Double>> skillMap        = new HashMap<>();
+
+    @Override
+    public void init(final int levelCap)
+    {
+        if (levelCap <= 1)
+        {
+            for (final Skill skill : Skill.values())
+            {
+                skillMap.put(skill, new Tuple<>(1, 0.0D));
+            }
+        }
+        else
+        {
+            final Random random = new Random();
+            for (final Skill skill : Skill.values())
+            {
+                skillMap.put(skill, new Tuple<>(random.nextInt(levelCap - 1) + 1, 0.0D));
+            }
+        }
+    }
+
+    @Override
+    public void init(@NotNull final ICitizenData mom, @NotNull final ICitizenData dad, final Random rand)
+    {
+        for (final Skill skill : Skill.values())
+        {
+            skillMap.put(skill, new Tuple<>((mom.getCitizenSkillHandler().getLevel(skill) + dad.getCitizenSkillHandler().getLevel(skill)) / 2 + rand.nextInt(CHILD_STATS_VARIANCE) - rand.nextInt(CHILD_STATS_VARIANCE), 0.0D));
```

Comment:
I think this is off now, would give a child nearly capped "experience" stats, maybe set a max amount they can inherit, like 5?

Reply:
Good that you mentioned it, yes.